src/effects/Saturation.cpp

```diff
@@ -102,6 +109,58 @@ std::shared_ptr<Frame> Saturation::GetFrame(std::shared_ptr<Frame> frame, int64_
 		G = constrain(G);
 		B = constrain(B);
 
+		/*
+		 * Color-separated saturation adjustment
+		 *
+		 * Splitting each of the three subpixels (R, G and B) into three distincs sub-subpixels (R, G and B in turn)
+		 * which in their optical sum reproduce the original subpixel's color OR produce white light in the brightness
+		 * of the original subpixel (dependening on the color channel's slider value).
```

Comment:
I would refer to these as _channels_ rather than "subpixels", I think, as it's a more relevant description of their function (algorithmically)... the fact that they _happen_ to correspond to LCD subpixels is really a coincidence, since the saturation function will work just fine if displayed on a CRT or a device with a non-RGB color system (as long as the RGB image data is mapped into that color space).

And if any other controls, such a CMY adjustments, were later added, then "subpixel" would no longer be even coincidentally accurate.

Reply:
Actually my documentation *intentionally* introduced the particular terms of sub-pixels and sub-sub-pixels to make clear that this is less about math and variables but more in fact about the repbuilding of single pixels using LEDs: Replace "sub-pixel" by "LED" and you see what the code actually does: For the "actual" sub-pixels of the LCD screen you can use a red, green and blue LED. For each of that LEDs, you can use another red, green and blue LED (the sub-sub-pixels). So the sole code and documentation is an instruction to rebuild the code using electronic devices -- and it would actually produce the same result as the code. So this is not about "happen to correspond", it is about the *idea* behind the code. And as the *idea* works so great, why drop it? Actually, when later adding CMY, the idea would still be working, as CMY does not exist as sub-sub-pixels (LEDs), it is just giving more power on G and B, B and R, and R and G.

Reply:
Okey-dokey. :+1: